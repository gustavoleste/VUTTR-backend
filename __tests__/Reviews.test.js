const { Reviews, connectDatabase } = require("../src/database/index");
const { databaseURLForTest } = require("../src/config");
const { userReview } = require("../src/helpers/index");
require("dotenv").config();

describe("Reviews Model", () => {
  beforeAll(async () => {
    await connectDatabase(databaseURLForTest, "reviewsModelTest");
  });

  afterAll(async () => {
    const database = await connectDatabase(
      databaseURLForTest,
      "reviewsModelTest"
    );
    await database.dropDatabase();
  });

  it("should not create a empty review", async () => {
    try {
      const newReview = new Reviews({});
      await newReview.save();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });

  it("should not create a review without the user field set", async () => {
    try {
      const reviewWithoutUserID = { ...userReview, user };
      const newReview = new Reviews(reviewWithoutUserID);
      await newReview.save();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });

  it("should not create a review without the tool field set", async () => {
    try {
      const reviewWithoutToolID = { ...userReview, tool };
      const newReview = new Reviews(reviewWithoutToolID);
      await newReview.save();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });

  it("should not create a review without the comment field set", async () => {
    try {
      const reviewWithoutComment = { ...userReview, comment };
      const newReview = new Reviews(reviewWithoutComment);
      await newReview.save();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });

  it("should create a new review", async () => {
    try {
      const newReview = new Reviews(userReview);
      await newReview.save();
      expect(newReview).toMatchSnapshot();
    } catch (error) {
      console.log(error);
    }
  });
});
