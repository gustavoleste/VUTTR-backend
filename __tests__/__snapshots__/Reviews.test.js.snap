// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews Model should create a new review 1`] = `
Object {
  "__v": 0,
  "_id": "9ceda7b37085d444ec1bec99",
  "comment": "Very useful tool",
  "createdAt": 2019-05-28T21:28:37.290Z,
  "tool": "4ceda7b37085d444ec1bec65",
  "updatedAt": 2019-05-28T21:28:37.290Z,
  "user": "1ceda7b37085d444ec1bec11",
}
`;

exports[`Reviews Model should not create a empty review 1`] = `[ValidationError: Reviews validation failed: comment: Path \`comment\` is required., tool: Path \`tool\` is required., user: Path \`user\` is required.]`;

exports[`Reviews Model should not create a review without the comment field set 1`] = `[ReferenceError: comment is not defined]`;

exports[`Reviews Model should not create a review without the tool field set 1`] = `[ReferenceError: tool is not defined]`;

exports[`Reviews Model should not create a review without the user field set 1`] = `[ReferenceError: user is not defined]`;
